drivers/axi_core/axi_pwmgen/axi_pwmgen.h

```diff
@@ -0,0 +1,134 @@
+/******************************************************************************/
+#include <stdint.h>
+#include <stdbool.h>
+
+/******************************************************************************/
+/*************************** Types Declarations *******************************/
+/******************************************************************************/
+/**
+ * @enum axi_pwmgen_polarity
+ * @brief Possible events for uart interrupt
+ */
+enum axi_pwmgen_polarity {
+	PWM_POLARITY_NORMAL,
+	PWM_POLARITY_INVERSED,
+};
+
+/**
+ * @struct axi_pwmgen_init_param
+ * @brief  Structure containing the init parameters needed by the PWM generator
+ */
+struct axi_pwmgen_init_param {
+	/** Base address where the PWM generator core is situated */
+	uint32_t base_addr;
+	/** PWM generator reference clock rate , in Hz */
+	uint32_t clk_rate_hz;
+	/** PWM generator period */
+	uint32_t period_ns;
+	/** PWM generator duty cycle */
+	uint32_t duty_cycle_ns;
+	/** PWM generator polarity */
+	enum axi_pwmgen_polarity polarity;
+	/** PWM generator enabled */
+	bool enabled;
+};
+
+/**
+ * @struct axi_pwmgen_desc
+ * @brief  Structure representing an PWM generator device
+ */
+struct axi_pwmgen_desc {
+	/** Base address where the PWM generator core is situated */
+	uint32_t base_addr;
+	/** PWM generator reference clock rate, in Hz */
+	uint32_t clk_rate_hz;
+	/** PWM generator period */
+	uint32_t period_ns;
+	/** PWM generator duty cycle */
+	uint32_t duty_cycle_ns;
+	/** PWM generator polarity */
+	enum axi_pwmgen_polarity polarity;
+	/** PWM generator enabled */
+	bool enabled;
+};
+
+/******************************************************************************/
+/************************ Functions Declarations ******************************/
+/******************************************************************************/
+/* Initialize the PWM generator device */
+int32_t axi_pwmgen_init(struct axi_pwmgen_desc **desc,
+			const struct axi_pwmgen_init_param *param);
+
+/* Free the resources used by the PWM generator device */
+int32_t axi_pwmgen_remove(struct axi_pwmgen_desc *desc);
+
+/* Enable PWM generator device */
+int32_t axi_pwmgen_enable(struct axi_pwmgen_desc *desc);
+
+/* Disable PWM generator device */
+int32_t axi_pwmgen_disable(struct axi_pwmgen_desc *desc);
+
+/* Set period of PWM generator device */
+int32_t axi_pwmgen_set_period(struct axi_pwmgen_desc *desc,
+			      uint32_t period_ns);
+
+/* Get period of PWM generator device */
+int32_t axi_pwmgen_get_period(struct axi_pwmgen_desc *desc,
+			      uint32_t *period_ns);
+
+/* Set duty cycle of PWM generator device */
+int32_t axi_pwmgen_set_duty_cycle(struct axi_pwmgen_desc *desc,
+				  uint32_t duty_cycle_ns);
+
+/* Get period of PWM generator device */
+int32_t axi_pwmgen_get_duty_cycle(struct axi_pwmgen_desc *desc,
+				  uint32_t *duty_cycle_ns);
+
+/* Set polarity of PWM generator device */
+int32_t axi_pwmgen_set_polarity(struct axi_pwmgen_desc *desc,
```

Comment:
If we have "get" for all the other functions, I would add it for "polarity" too.